find_attitude/constraints.py

```diff
@@ -0,0 +1,132 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import functools
+import logging
+from dataclasses import dataclass
+
+import astropy.units as u
+import astropy_healpix
+import numpy as np
+import ska_sun
+from cxotime import CxoTime, CxoTimeLike
+from Quaternion import Quat, QuatLike
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class Constraints:
+    """Constraints on attitude for attitude solution.
+
+    Parameters
+    ----------
+    att : QuatLike or None
+        Estimated attitude, assumed to be within ``att_err`` of true attitude.
+    att_err : float
+        Radial uncertainty of the attitude in degrees (default=4.0).
+    pitch : float | None
+        If not None, constrain sun pitch angle to ``pitch +/- pitch_err`` degrees.
+        Default is None => no pitch constraint.
+    pitch_err : float
+        Uncertainty in pitch in degrees (default=1.5).
+    off_nom_roll_max : float or None
+        Maximum off-nominal roll angle in degrees (default=2.0).
+    min_stars : int or None
+        Minimum number of stars required for a valid solution. Default is None, meaning
+        the minimum number of stars is determined from the constraints. Key use case for
```

Comment:
Regarding "Default is None, meaning the minimum number of stars is determined from the constraints." - I don't understand what this means.  This is the constraint so how is the minimum number of stars determined from the constraints?

Reply:
I suggest a reference to see get_min_stars() .